backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/tutor/user/service/DeleteExternalInactiveUsersTest.groovy

```
@@ -0,0 +1,94 @@
+package pt.ulisboa.tecnico.socialsoftware.tutor.user.service
+
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
+import org.springframework.boot.test.context.TestConfiguration
+import pt.ulisboa.tecnico.socialsoftware.tutor.BeanConfiguration
+import pt.ulisboa.tecnico.socialsoftware.tutor.SpockTest
+import pt.ulisboa.tecnico.socialsoftware.tutor.course.Course
+import pt.ulisboa.tecnico.socialsoftware.tutor.course.CourseExecution
+import pt.ulisboa.tecnico.socialsoftware.tutor.user.User
+
+@DataJpaTest
+class DeleteExternalInactiveUsersTest extends SpockTest{
+
+    def user1
+    def user2
+
+    List userIdList
+
+    def setup() {
+        course = new Course(COURSE_1_NAME, Course.Type.EXTERNAL)
+        courseRepository.save(course)
+
+        courseExecution = new CourseExecution(course, COURSE_1_ACRONYM, COURSE_1_ACADEMIC_TERM, Course.Type.EXTERNAL)
+        courseExecutionRepository.save(courseExecution)
+
+        userIdList = new ArrayList<Integer>();
+    }
+
+
+
+    def "tries to delete an active user" () {
+        given: "one active users"
+        user1 = new User(USER_1_NAME, USER_1_USERNAME, User.Role.STUDENT)
+        user1.setState(User.State.ACTIVE)
+        userRepository.save(user1)
+        courseExecution.addUser(user1)
+        user1.addCourse(courseExecution)
+
+        and: "a userId list"
+        userIdList << user1.getId()
+
+        when:
+        userService.deleteExternalInactiveUsers(userIdList);
+
+        then: "the user is not removed from his course execution"
```

if the user is active it should throw an exception